tests/kani/AsyncAwait/spawn.rs

```diff
@@ -0,0 +1,30 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+//
+// compile-flags: --edition 2018
+
+//! This file tests the executor and spawn infrastructure from the Kani library.
+
+use std::sync::{
+    atomic::{AtomicI64, Ordering},
+    Arc,
+};
+
+#[kani::proof]
+#[kani::unwind(4)]
+fn deterministic_schedule() {
```

Comment:
This is not necessarily testing that the schedule is deterministic though.

Reply:
Now, but it tests the behavior under a deterministic schedule. I'm not sure what you think should be improved.

Reply:
I want to make sure what we are shipping in the Kani crate follows its spec. There is no test in this PR that actually ensures that the RoundRobin scheduler is in fact RoundRobin.

These tests are internal to the tool, and adding extra checks here doesn't mean we have to always keep the same order. But we ensure that our implementation is correct.

Also, if we do pick one specific order, that should be clear to the user.

Reply:
Alright, I'll add asserts to check this and change the name of the harness